server/src/services/metadata.service.ts

```diff
@@ -162,6 +162,14 @@ export class MetadataService extends BaseService {
 
     this.logger.verbose('Exif Tags', exifTags);
 
+    if (!asset.fileCreatedAt) {
+      asset.fileCreatedAt = stats.mtime;
+    }
+
+    if (!asset.fileModifiedAt) {
+      asset.fileModifiedAt = stats.mtime;
+    }
```

Comment:
What's this for?

Reply:
External assets :)

Since they are imported by the library service with null dates, this is where they become non-null.